test/api/importcsv/api_test.go

```diff
@@ -0,0 +1,166 @@
+			for _, imp := range outputImportSummaries {
+				if uint(imp.ID) == inputData.ID {
+					outputMatchingSummary = imp
+				}
+			}
+			assert.FlatEqual(len(expectedApps)+len(expectedDeps), outputMatchingSummary.ValidCount)
+
+			// inject import summary id into Summary root
+			pathForImportSummary := binding.Path(api.SummaryRoot).Inject(binding.Params{api.ID: inputID})
+
+			// Get summaries of the Input ID.
+			outputImportSummary := api.ImportSummary{}
+			assert.Should(t, Client.Get(pathForImportSummary, &outputImportSummary))
+
+			// Get all imports.
+			var outputImports []api.Import
+			assert.Should(t, Client.Get(api.ImportsRoot, &outputImports))
+			j, k := 0, 0
+			for _, imp := range outputImports {
+				if imp["recordType1"] == 1 && j < len(expectedApps) {
+					// An Application with no dependencies.
+					if expectedApps[j].Name != imp["applicationName"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].Name, imp["applicationName"])
+					}
+					if expectedApps[j].Description != imp["description"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].Description, imp["description"])
+					}
+					if expectedApps[j].BusinessService.Name != imp["businessService"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedApps[j].BusinessService.Name, imp["businessService"])
+					}
+					j++
+				}
+				if imp["recordType1"] == 2 && k < len(expectedDeps) {
+					// An Application with Dependencies.
+					if expectedDeps[k].From.Name != imp["applicationName"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedDeps[k].From.Name, imp["applicationName"])
+					}
+					if expectedDeps[k].To.Name != imp["dependency"] {
+						t.Errorf("Mismatch in name of import: Expected %s, Actual %s", expectedDeps[k].To.Name, imp["dependency"])
+					}
+					k++
+				}
+			}
+
+			// Download the csv.
+			pathToGotCSV := "downloadcsv.csv"
+			err := Client.FileGet(api.DownloadRoot, pathToGotCSV)
+			if err != nil {
+				t.Errorf("Error downloading CSV: %v", err)
+			}
+
+			// Read the got CSV file.
+			gotCSV, err := ioutil.ReadFile(pathToGotCSV)
+			if err != nil {
+				t.Errorf("Error reading CSV: %s", pathToGotCSV)
+			}
+			gotCSVString := string(gotCSV)
+
+			// Read the expected CSV file.
+			expectedCSV, err := ioutil.ReadFile(r.FileName)
+			if err != nil {
+				t.Errorf("Error reading CSV: %s", r.FileName)
+			}
+			expectedCSVString := string(expectedCSV)
+
+			if gotCSVString != expectedCSVString {
+				t.Errorf("The CSV files have different content %s and %s", gotCSVString, expectedCSVString)
+			}
+
+			// Remove the CSV file created.
+			err = os.Remove(pathToGotCSV)
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+
+			// Delete import summary
+			assert.Should(t, Client.Delete(pathForImportSummary))
+
+			// Delete all imports
+			id := 1
```

Comment:
This is not the right way - expecting ID of objects there are created by client (not seeded by the application). I'd suggest skip the Imports deletetion and be OK with ImportSummary delete (that should take imports too).